stdlib/public/TensorFlow/Dataset.swift

```diff
@@ -78,30 +70,37 @@ extension SingleValueDataset : Sequence {
 
 public extension SingleValueDataset {
   @inlinable @inline(__always)
-  func map<T>(
-    _ transform: @convention(tensorflow) (Tensor<Scalar>) -> Tensor<T>
-  ) -> SingleValueDataset {
-    // A dataset creation op only runs on TF CPU.
-    enableCPU()
-    return SingleValueDataset(
+  func map<T : AccelerableByTensorFlow>(
+    _ transform: @convention(tensorflow) (Tensor<ScalarOfElement>) -> Tensor<T>
+  ) -> SingleValueDataset<T> {
+    // FIXME: If we pass an empty Array<TensorHandle<T>> as other_arguments and
```

Comment:
Plz file a bug and cc me on it if you haven't already (I'm behind on email)

Reply:
https://bugs.swift.org/browse/SR-8570